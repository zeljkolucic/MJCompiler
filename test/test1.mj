program MJProgram
	int x, y, const int z = 3;
{
	int sum(int x, y) {
		return x + y;
	}
	
	void main() {
	
		x = 0;
		y = 1;
		
		if(x + > 3) {
			return;
		}
		
		x = y++ + 3 +;
		
	}
}