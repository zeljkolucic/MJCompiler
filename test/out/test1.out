Linija [5]: Pristup formalnom parametru x.
DesignatorIdent(
 x
) [DesignatorIdent]
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog MJProgram: notype, 0, 1 
   Meth sum: int, 0, 1 
      Var x: int, 0, 1 

   Meth main: notype, 0, 0 



